fluent/syntax/parser.py

```diff
@@ -349,13 +349,14 @@ def get_number(self, ps):
     @with_span
     def get_pattern(self, ps):
         elements = []
-        ps.skip_ws()
+        ps.skip_inline_ws()
 
         while ps.current():
             ch = ps.current()
 
             if ch == '\n':
-                break
+                if not ps.is_peek_next_line_pattern():
+                    break
```

Comment:
I think returning `None` as an option is fine. If you really want to avoid that, it seems that returning a `TextElement` with an empty buffer and then testing against the length of the buffer on the last element also works.
I think I prefer that over double `is_peek_next_line_pattern`, but quite honestly, not by that much. If you find the duplication easier, I'm ok with it.